packages/plugin-json/src/lib/amplience-schema-transformers.ts

```diff
@@ -47,58 +50,60 @@ export const contentTypeSchemaBody = (
   type: ObjectTypeDefinitionNode,
   schema: GraphQLSchema,
   schemaHost: string,
-): AmplienceContentTypeSchemaBody => {
-  const properties = objectProperties(type, schema, schemaHost);
-  const propertyOrder =
-    type.fields
-      ?.filter(
-        (field) =>
-          isAmplienceProperty(type, field) && !hasDeliveryKeyDirective(field),
-      )
-      .map((field) => field.name.value) ?? [];
+): AmplienceContentTypeSchemaBody => ({
+  $id: typeUri(type, schemaHost),
+  $schema: "http://json-schema.org/draft-07/schema#",
+  type: "object",
+  ...refType(AMPLIENCE_TYPE.CORE.Content),
 
-  const deliveryKeyDirective = maybeDeliveryKeyDirective(type);
-  if (deliveryKeyDirective) {
-    properties._meta = deliveryKeyMetaProperty(deliveryKeyDirective);
-    // always place delivery keys at the top of the form, if present
-    propertyOrder.unshift("_meta");
-  }
+  title: capitalCase(type.name.value),
 
-  return {
-    $id: typeUri(type, schemaHost),
-    $schema: "http://json-schema.org/draft-07/schema#",
-    ...refType(AMPLIENCE_TYPE.CORE.Content),
-    title: capitalCase(type.name.value),
-    properties,
-    description: type.description?.value ?? capitalCase(type.name.value),
-    "trait:sortable": sortableTrait(type),
-    "trait:hierarchy": isHierarchy(type)
-      ? hierarchyTrait(type, schema, schemaHost)
-      : undefined,
-    "trait:filterable": filterableTrait(type),
-    type: "object",
-    propertyOrder,
-    required: type.fields
+  properties: {
+    // only add _meta property for delivery key
+    ...ifValue(maybeDeliveryKeyDirective(type), (directive) => ({
+      _meta: deliveryKeyMetaProperty(directive),
+    })),
+    ...objectProperties(type, schema, schemaHost),
+  },
+  description: type.description?.value ?? capitalCase(type.name.value),
+
+  "trait:sortable": sortableTrait(type),
+  "trait:hierarchy": isHierarchy(type)
+    ? hierarchyTrait(type, schema, schemaHost)
+    : undefined,
+  "trait:filterable": filterableTrait(type),
+
+  propertyOrder: [
+    // always place delivery keys at the top
+    maybeDeliveryKeyDirective(type) ? "_meta" : undefined,
+
+    ...(type.fields
       ?.filter(
         (field) =>
           isAmplienceProperty(type, field) && !hasDeliveryKeyDirective(field),
       )
-      .filter(
-        (field) =>
-          field.type.kind === "NonNullType" ||
-          hasDirective(field, "amplienceLocalized"),
-      )
-      .map((n) => n.name.value),
-    ...(isHierarchy(type)
-      ? {
-          allOf: [
-            refType(AMPLIENCE_TYPE.CORE.Content).allOf[0],
-            refType(AMPLIENCE_TYPE.CORE.HierarchyNode).allOf[0],
-          ],
-        }
-      : {}),
-  };
-};
+      .map((field) => field.name.value) ?? []),
+  ].filter(isValue),
+
+  required: type.fields
+    ?.filter((field) => isAmplienceProperty(type, field))
```

Comment:
While validation does prevent us from ending up with a required deliveryKey in theory, I think it'd be good to add a `!hasDeliveryKey(field)` here just for absolute certainty in case something changes elsewhere. Required keys prevent content items from being copied in Amplience so it's critical that this never happens